compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -1004,25 +1004,49 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
     fn lower_assoc_ty_constraint(
         &mut self,
         constraint: &AssocTyConstraint,
-        itctx: ImplTraitContext<'_, 'hir>,
+        mut itctx: ImplTraitContext<'_, 'hir>,
     ) -> hir::TypeBinding<'hir> {
         debug!("lower_assoc_ty_constraint(constraint={:?}, itctx={:?})", constraint, itctx);
 
-        if let Some(ref gen_args) = constraint.gen_args {
-            self.sess.span_fatal(
-                gen_args.span(),
-                "generic associated types in trait paths are currently not implemented",
-            );
-        }
+        // lower generic arguments of identifier in constraint
+        let gen_args = if let Some(ref gen_args) = constraint.gen_args {
+            let gen_args_ctor = match gen_args {
+                GenericArgs::AngleBracketed(ref data) => {
```

Comment:
Yes.

Reply:
I mean, it changes a nested if/if and else to a single match; that definitely seems like a style improvement.

Reply:
It could certainly be a style improvement, but I think from a logical perspective it makes more sense if we keep it this way. We either allocate from the arena or we don't, which depends on whether generic arguments are available or not. The distinction between the kind of generic arguments is logically on a different level. But I think it's a nit.